src/batched/KokkosBatched_InverseLU_Serial_Impl.hpp

```diff
@@ -0,0 +1,195 @@
+        //                     (double*)A.data(), A.stride_1(), 
+        //                     (MKL_INT*)&r_val, format, (MKL_INT)vector_type::vector_length);
+
+        //void mkl_dgetrfnp_compact (MKL_LAYOUT layout, MKL_INT m, MKL_INT n, double * ap, MKL_INT ldap, 
+        //                           MKL_INT * info, MKL_COMPACT_PACK format, MKL_INT nm);
+
+        //void mkl_dgetrinp_compact (MKL_LAYOUT layout, MKL_INT n, double * ap, MKL_INT ldap, double * work, 
+        //                           MKL_INT lwork, MKL_INT * info, MKL_COMPACT_PACK format, MKL_INT nm);
+        mkl_dgetrinp_compact (MKL_COL_MAJOR, n, 
+                              (double*)A.data(), A.stride_1(), 
+                              (double*)W.data(), n*n, 
+                              (MKL_INT*)&r_val, format, (MKL_INT)vector_type::vector_length);
+
+      } else if (A.stride_1() == 1) {
+        //mkl_dgetrfnp_compact(MKL_ROW_MAJOR, 
+        //                     m, n, 
+        //                     (double*)A.data(), A.stride_0(), 
+        //                     (MKL_INT*)&r_val, format, (MKL_INT)vector_type::vector_length);
+        mkl_dgetrinp_compact (MKL_ROW_MAJOR, n, 
+                              (double*)A.data(), A.stride_0(), 
+                              (double*)W.data(), n*n, 
+                              (MKL_INT*)&r_val, format, (MKL_INT)vector_type::vector_length);
+      } else {
+        r_val = -1;
+      }
+      return r_val;
+    }
+#endif
+
+    template<>
+    template<typename AViewType,
+             typename WViewType>
+    KOKKOS_INLINE_FUNCTION
+    int
+    SerialInverseLU<Algo::InverseLU::Unblocked>::
+    invoke(const AViewType &A, 
+           const WViewType &W,
+           const typename MagnitudeScalarType<typename AViewType::non_const_value_type>::type tiny) {
+        static_assert(AViewType::rank == 2, "A should have two dimensions");
+        static_assert(std::is_same<typename AViewType::memory_space, typename WViewType::memory_space>::value, "A and W should be on the same memory space");
+        assert(A.span()*sizeof(typename AViewType::value_type) <= W.span()*sizeof(typename WViewType::value_type));
+
+        typedef typename AViewType::value_type ScalarType;
+
+        int B_stride_0, B_stride_1;
+        if (WViewType::rank == 1) {
+            if (std::is_same<typename WViewType::array_layout, Kokkos::LayoutRight>::value){
+                B_stride_0 = A.extent(1);
+                B_stride_1 = 1;
+            } 
+            else {
+                B_stride_0 = A.stride_0();
+                B_stride_1 = A.stride_0()*A.extent(0);
+            }
+        }
+        else { //WViewType::rank = 2
+            B_stride_0 = A.stride_0();
+            B_stride_1 = A.stride_1();
+        }
+        auto B = Kokkos::View<ScalarType**, Kokkos::LayoutStride, typename WViewType::memory_space, Kokkos::MemoryTraits<Kokkos::Unmanaged> >(W.data(), A.extent(0), B_stride_0, A.extent(1), B_stride_1);
```

Comment:
Does this constructor really work ? I have not seen this kind of constructor taking strides as inputs. Could you put a link to kokkos file and line ?

Reply:
@kyungjoo-kim Yes it works.
https://github.com/kokkos/kokkos/blob/2169d23dab6ddbb7993ad68104e8a0c6c44bab84/core/src/Kokkos_View.hpp#L2296-L2321
If it is LayoutStride, the Layout constructor can take strides as inputs:
https://github.com/kokkos/kokkos/blob/2169d23dab6ddbb7993ad68104e8a0c6c44bab84/core/src/Kokkos_Layout.hpp#L180-L193 

To make it more sense, I change it to:
```auto B = Kokkos::View<ScalarType**, Kokkos::LayoutStride, typename WViewType::memory_space, Kokkos::MemoryTraits<Kokkos::Unmanaged> >(W.data(), Kokkos::LayoutStride(A.extent(0), B_stride_0, A.extent(1), B_stride_1));```

which uses this constructor:
https://github.com/kokkos/kokkos/blob/2169d23dab6ddbb7993ad68104e8a0c6c44bab84/core/src/Kokkos_View.hpp#L2323-L2330

Reply:
BTW, how can I show the codes instead of links?

Reply:
@vqd8a If you right-click on a line of code in GitHub, you can get a permanent link to it.